superset/assets/spec/javascripts/explore/utils_spec.jsx

```diff
@@ -128,6 +131,56 @@ describe('exploreUtils', () => {
     });
   });
 
+  describe('domain sharding', () => {
+    let stub;
+    const availableDomains = [
+      'http://localhost/',
+      'domain1.com', 'domain2.com', 'domain3.com',
+    ];
+    beforeEach(() => {
+      stub = sinon.stub(hostNamesConfig, 'availableDomains').value(availableDomains);
+    });
+    afterEach(() => {
+      stub.restore();
+    });
+
+    it('generate url to different domains', () => {
+      let url = getExploreUrlAndPayload({
+        formData,
+        endpointType: 'json',
+        allowDomainSharding: true,
+      }).url;
+      expect(url).toMatch(availableDomains[0]);
+
+      url = getExploreUrlAndPayload({
+        formData,
+        endpointType: 'json',
+        allowDomainSharding: true,
+      }).url;
+      expect(url).toMatch(availableDomains[1]);
+
+      url = getExploreUrlAndPayload({
```

Comment:
is it worthwhile to add one more to make sure it goes back to `availableDomains[0]`?

Reply:
added 2 more...